internal/storage/v2/grpc/tracereader_test.go

```diff
@@ -22,21 +22,36 @@ import (
 type testServer struct {
 	storage.UnimplementedTraceReaderServer
 
-	getServicesError error
+	err error
 }
 
 func (s *testServer) GetServices(
 	context.Context,
 	*storage.GetServicesRequest,
 ) (*storage.GetServicesResponse, error) {
-	if s.getServicesError != nil {
-		return nil, s.getServicesError
+	if s.err != nil {
+		return nil, s.err
 	}
 	return &storage.GetServicesResponse{
 		Services: []string{"service-a", "service-b"},
 	}, nil
 }
 
+func (s *testServer) GetOperations(
+	context.Context,
+	*storage.GetOperationsRequest,
+) (*storage.GetOperationsResponse, error) {
+	if s.err != nil {
+		return nil, s.err
+	}
+	return &storage.GetOperationsResponse{
+		Operations: []*storage.Operation{
```

Comment:
This separation of return values from inputs makes the tests harder to read / validate. I would rather have a field in the struct for return value + error and return that pair unconditionally here. Then in the test you can provide outputs, similar to how a regular mock would work.